src/components/project/list.vue

```diff
@@ -95,7 +95,15 @@ export default {
   setup() {
     const { currentUser, projects } = useRequestData();
 
-    const sortMode = ref('latest');
+    const sortMode = computed({
+      get() {
+        return currentUser.preferences.site.projectSortMode || 'latest';
+      },
+      set(val) {
+        currentUser.preferences.site.projectSortMode = val;
+      },
+    });
```

Comment:
I think there's a subtle issue with this setup. I took a look at the failing test in CircleCI just to see what it was, and I saw the error:

```
TypeError: Cannot read properties of undefined (reading 'site')
```

I'm pretty sure this is the only part of your code that accesses a `site` property, which points to the fact that `currentUser.preferences` is somehow becoming `undefined` before this code is run. Looking at the test, the test is related to logout. One thing that logout does is reset `requestData`, including `currentUser`. Once `currentUser` is reset, `currentUser.preferences` will be `undefined`.

I don't think this is just happening in testing. When I try things out locally, I also see an error when I try to log out from the homepage.

What's weird is that logging out also navigates away from the route, which will unmount the `ProjectList` component. The component should be unmounted in the same tick that `requestData` is cleared. But if the component is being unmounted, that means it isn't patching the DOM, so why is it even trying to access this computed ref that is used in the template?

I think the answer is that a `watchEffect()` below reads `sortFunction`, which reads `sortMode`, which reads `currentUser`. The change to `currentUser` triggers the `watchEffect()`, which I guess is rerun even though the component is about to be unmounted.

I think one way to get around this is to weaken the link between `sortMode` and `currentUser`. If `sortMode` isn't defined based on `currentUser`, then resetting `currentUser` won't change `sortMode`, so it won't trigger the `watchEffect()`. You could change `sortMode` back to being a ref by doing something like this:

```js
const sortMode = ref(currentUser.preferences.site.projectSortMode || 'latest');
watch(sortMode, (val) => {
  currentUser.preferences.site.projectSortMode = val;
});
```

That way, `currentUser` will be used to _initialize_ `sortMode`, but that's it. When `currentUser` is reset later, it won't affect `sortMode`, because `sortMode` has already been initialized. When I make this code change, the test passes.

Sorry, this is a bit of a weird one! But the good news is that this isn't a common issue that comes up.